Feat: getLogs indexer update

0xpeluche opened the pull request:
> Previously, v2 had an internal API logic that rejected getLogs requests without a target when the blockRange exceeded 10k blocks, to prevent overloading database queries that would scan a large portion of rows without target. Added logic to batch queries without a target when the blockRange exceeds 10k blocks, and implemented a systematic rejection of queries without a target when the blockRange exceeds 500k blocks. This prevents sending 50 batches to the API when no target is specified, thereby avoiding API/database overload

g1nt0ki commented on src/util/indexer.ts:
```diff
@@ -365,53 +370,119 @@ export async function getLogs({ chain = 'ethereum', topic, topics, fromBlock, to
+    const batches = Math.ceil(blockRange / BATCH_SIZE)
+    const MAX_CONCURRENT_BATCHES = 5
+    
+    for (let i = 0; i < batches; i += MAX_CONCURRENT_BATCHES) {
```
I feel safer if we use `@supercharge/promise-pool`  for tracking concurrency 

Also, probably, we can do it sequentially for now to simplify implementation?

g1nt0ki reviewed (commented):
.

0xpeluche commented:
- Code harmonization between getLogs (indexer) and getLogs (RPCs - Wrapper)
- Removed v1 to strictly use v2 indexers, EXCEPT for the /balances endpoint
- Refactored and deduplicated code
